dart-processor/src/main/java/com/f2prateek/dart/processor/InjectExtraProcessor.java

```diff
@@ -78,9 +78,13 @@
 
     // Try to find a parent injector for each injector.
     for (Map.Entry<TypeElement, InjectionTarget> entry : targetClassMap.entrySet()) {
-      String parentClassFqcn = findParentFqcn(entry.getKey(), erasedTargetTypes);
+      TypeElement typeElement = entry.getKey();
+      InjectionTarget injectionTarget = entry.getValue();
+      String parentClassFqcn = findParentFqcn(typeElement, erasedTargetTypes);
       if (parentClassFqcn != null) {
-        entry.getValue().setParentTarget(parentClassFqcn);
+        injectionTarget.setParentTarget(parentClassFqcn);
+      } else {
+        findInjectionTargetForSuperTypeWithInjectExtra(typeElement, injectionTarget);
```

Comment:
are we using ancestors besides the parent?

Reply:
You're right. It's complicated because for Henson we want all parts, and darts only need one.
For henson in DH3 we solved this and it only needs the closest parent... I got to rework on that, I think both processes can't be unified.